src/lib/holocene/split-button.svelte

```diff
@@ -1,47 +1,56 @@
 <script lang="ts">
   import Icon from '$holocene/icon/index.svelte';
-  import MenuContainer from './primatives/menu/menu-container.svelte';
-  import MenuButton from './primatives/menu/menu-button.svelte';
-  import Menu from './primatives/menu/menu.svelte';
+  import MenuContainer from '$holocene/primitives/menu/menu-container.svelte';
+  import MenuButton from '$holocene/primitives/menu/menu-button.svelte';
+  import Menu from '$holocene/primitives/menu/menu.svelte';
+  import Button from './button.svelte';
+
   export let label: string;
   export let id: string;
   export let disabled: boolean = false;
   export let left: boolean = false;
   export let right: boolean = false;
+  export let href = '';
 
   let show: boolean = false;
 </script>
 
 <MenuContainer class={$$props.class}>
-  <MenuButton bind:show controls={id} class={$$props.class} {disabled} on:click>
-    <div class="split-button" class:disabled>
-      <button tabindex="-1" {disabled} class="segment rounded-l px-4">
-        {label}
-      </button>
-      <div class="segment rounded-r px-2">
-        <Icon stroke="currentcolor" name="caretDown" />
-      </div>
-    </div>
-  </MenuButton>
-  <Menu class="split-button-menu" {id} {show} {left} {right}>
+  <div class="split-button" class:disabled>
+    <Button {href} variant="primary" {disabled} class="segment left" on:click>
+      {label}
+    </Button>
+    <MenuButton dark class="segment right" bind:show controls={id} {disabled}>
+      <Icon stroke="currentcolor" name="caretDown" />
+    </MenuButton>
+  </div>
+  <Menu class="min-w-max" {id} {show} {left} {right}>
     <slot />
   </Menu>
 </MenuContainer>
 
 <style lang="postcss">
   .split-button {
     @apply flex grow cursor-pointer flex-row gap-[1px] font-secondary;
-  }
 
-  .split-button.disabled {
-    @apply cursor-not-allowed opacity-50;
-  }
+    :global(.segment) {
+      @apply relative flex w-fit items-center justify-center py-2 font-secondary text-sm;
+    }
 
-  :global(.split-button-menu) {
-    @apply flex min-w-max flex-col items-start gap-y-4 border-gray-300 p-3 px-6 font-secondary text-sm;
-  }
+    :global(.segment.left) {
```

Comment:
I was thinking on these you could do the terrible thing of :first-child and :last-child and then the css becomes automagic for any amount of containing elements without having to ho hum about what is the leftmost part of the button and the right most. This is totally fine too just a thought

Reply:
Ah yeah, I think for split button this should be OK, but for something like the Toggle Buttons, we will definitely want to do that.
<img width="329" alt="Screen Shot 2022-07-27 at 10 37 41 AM" src="https://user-images.githubusercontent.com/11775628/181302069-6c86017b-3e5c-44b5-a5f5-9f39f082f0a1.png">

Reply:
Yeah when I think about it those feel like they could/should be composed of the more primative 🦍 elements. But this approach is also fine for right meow